execution/function/histogram.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
+
+package function
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"sync"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/thanos-community/promql-engine/execution/model"
+)
+
+type histogramSeries struct {
+	outputID   int
+	upperBound float64
+}
+
+// functionOperator returns []model.StepVector after processing input with desired function.
+type histogramOperator struct {
+	pool *model.VectorPool
+
+	funcArgs parser.Expressions
+
+	once     sync.Once
+	series   []labels.Labels
+	scalarOp model.VectorOperator
+	vectorOp model.VectorOperator
+
+	// scalarPoints is a reusable buffer for points from the first argument of histogram_quantile.
+	scalarPoints []float64
+	// outputIndex is a mapping from input series ID to the output series ID and its upper boundary value
+	// parsed from the le label.
+	outputIndex []histogramSeries
+	// seriesBuckets are the buckets for each individual series.
+	seriesBuckets []buckets
+}
+
+func NewHistogramOperator(pool *model.VectorPool, args parser.Expressions, nextOps []model.VectorOperator, stepsBatch int) (model.VectorOperator, error) {
+	return &histogramOperator{
+		pool:         pool,
+		funcArgs:     args,
+		once:         sync.Once{},
+		scalarOp:     nextOps[0],
+		vectorOp:     nextOps[1],
+		scalarPoints: make([]float64, stepsBatch),
+	}, nil
+}
+
+func (o *histogramOperator) Explain() (me string, next []model.VectorOperator) {
+	next = []model.VectorOperator{o.scalarOp, o.vectorOp}
+	return fmt.Sprintf("[*functionOperator] %v(%v)", "histogram_quantile", o.funcArgs), next
```

Comment:
We don't need `%v` for `histogram_quantile`

Reply:
Umm, actually I meant we can remove `%s` totally since `histogram_quantile` is a constant :)